tests/bdp_spark.erl

```diff
@@ -0,0 +1,138 @@
+    lager:info("Service definition ~p (~s) added to node ~p", [ServiceName, ?SPARK_MASTER_TYPE, Node]),
+    ok.
+
+
+start_services(Node, [{ServiceNode, ServiceName} | Rest]) ->
+    ok = bdp_util:service_started(Node, ServiceNode, ServiceName, ?SPARK_MASTER_TYPE),
+    lager:info("Service ~p up on ~p node", [ServiceName, ServiceNode]),
+    start_services(Node, Rest);
+start_services(_, []) -> ok.
+
+
+stop_services(Node, [{ServiceNode, ServiceName} | Rest]) ->
+    ok = bdp_util:service_stopped(Node, ServiceNode, ServiceName, ?SPARK_MASTER_TYPE),
+    lager:info("Service ~p down on ~p", [ServiceName, Node]),
+    stop_services(Node, Rest);
+stop_services(_, []) -> ok.
+
+
+remove_spark_service(Node, ServiceName, WaitServices) ->
+    ok = bdp_util:service_removed(Node, ServiceName),
+    ok = bdp_util:wait_services(Node, {[], WaitServices}),
+    lager:info("Service ~p removed from ~p", [ServiceName, Node]),
+    ok.
+
+
+test_spark_fail_recovery() ->
+    %% execute $ ps -ef | grep org.apache.spark.deploy.master.Master
+    %% to find out the PIDs of running masters and to make sure they are actually running
+    %%
+    %% To find out which of the masters is the leader open spark
+    %% master log files located in the spark-master/logs if the line
+    %%
+    %% ‘INFO RiakEnsembleLeaderElectionAgent: We have lost leadership’
+    %%
+    %% is found as a last one from RiakEnsembleLeaderElectionAgent,
+    %% this instance is currently a standby master. Let’s ‘tail -f’
+    %% this log.  This way we’ll see when it’ll gain leadership
+
+    %% Go to the instance that has and entry
+    %% ‘INFO RiakEnsembleLeaderElectionAgent: We have gained leadership’
+    %% and execute ‘kill <master’s pid>’
+
+    %% Verify that in the standby master’s logs that you tail there are entries like
+    %% 15/07/09 02:13:46 INFO RiakEnsembleLeaderElectionAgent: We have gained leadership
+    %% 15/07/09 02:13:46 INFO Master: I have been elected leader! New state: RECOVERING
+
+    [DevPath | _] = rtdev:devpaths(),
+
+    Node1Path = DevPath ++ "/dev/dev" ++ integer_to_list(1),
+    Node2Path = DevPath ++ "/dev/dev" ++ integer_to_list(2),
+
+    lager:info("Node1 path = ~s, Node2 path = ~s", [Node1Path, Node2Path]),
+
+    LogSubPath = "/lib/data_platform-1/priv/spark-master/logs/spark-" ++ ?SPARK_IDENT_STRING ++ "-org.apache.spark.deploy.master.Master-1-" ++ ?HOSTNAME ++  ".out",
+    Spark1LogFile = Node1Path ++ LogSubPath,
+    Spark2LogFile = Node2Path ++ LogSubPath,
+
+    lager:info("Spark service one log path ~s", [Spark1LogFile]),
+    lager:info("Spark service two log path ~s", [Spark2LogFile]),
+ 
+    os:cmd("chmod +x ./priv/bdp_spark_test/leader_election_check.sh"),
+    Command = "./priv/bdp_spark_test/leader_election_check.sh " ++ Spark1LogFile ++ " " ++ Spark2LogFile ++ " " ++ ?TIMEOUT,
+    lager:info("Running bash script: ~s", [Command]),
+    Res = os:cmd(Command),
+    lager:info("Res= ~s", [Res]),
+    ?assert(Res == "ok\n"),
```

Comment:
Using `?assertEquals("ok\n", Res)` instead of a raw assert means if the assert fails, it will print both the expected and received values. This means the log statement above is unneeded; in the passing case we know what it is and don't need it output, and in the failing case, the value is output.